UnityProject/Assets/Scripts/Objects/Closets/AbandonedCrates.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections.Generic;
+using MiniGames;
+using Mirror;
+using UnityEngine;
+
+namespace Objects.Closets
+{
+	public class AbandonedCrates : ClosetControl, IMiniGame
```

Comment:
It doesn't make much sense for a minigame to _be_ a closet. Rather, a minigame can _have_ a closet.

`GetComponent<ClosetControl>.DoThingsToClosetControl()` would be more appropriate. This would also allow this class to inherit from `MiniGameModule`, if that makes more sense to you.

Reply:
The minigame is not the closet, AbandonedCrate is a closet that has it's own behavior.

Reply:
All of this code is about a minigame, not a closet.

Reply:
Like I said, Abandoned Crates are closets with their own behavior. Their behavior is that they use minigames to be opened which in this context makes sense because we want the closets' behavior for storing objects and closing/opening them. The minigames take place the open/close interactions and that's the whole point of those crates, minigames.

Am I misreading something?

Reply:
The problem is this can't be easily used for any type of `ClosetControl`. Sure, we might only have one `ClosetControl` script for now, but if we add others, then we could no longer use this script and we'd need to refactor it, if we wanted to add minigames to them. However, we use _composition_,, we don't need to worry about what the `ClosetControl` looks like, just that we trust it to do its job.

So, I am indicating a design issue. But I won't fight you on it.

Reply:
You actually make a really good point, but:
The problem with separating this into two scripts is that Unity has a problem with component hierarchies, which have caused me headaches multiples times before while working on stuff that involves interactions. So, while I do agree with you, I don't want this to randomly break in the future because Unity decided to shuffle component positions around.

Reply:
I still believe this could very well just be its own component and use a reference to Closet control. Don't see why it should extend it.

Reply:
I agree with Gilles here.

Reply:
I feel like almost everything here belongs into the minigame tracker.

Reply:
can you explain to me how I can move this to the minigame tracker? I don't see how It makes sense to be moved there.

Reply:
also, again, read what I said about interactions. How do you want to trigger the minigame if you guys didn't like me putting it inside the closet code and I have told you already how having two interactable interfaces causes one to be priortized over the other.

Reply:
I thought a minigame was supposed to be generic enough to be used for multiple things. I reckon the minigame tracker (maybe call it a manager) would receive requests for a minigame, start the minigame and handle the minigame success/fail.
Why does the abandoned crate check for multiple minigame modules? It should not need to care about that.
About the interaction: We want to react when a player tries to open the locked crate. Why not create an event that fires if it's locked, which is listened to by a minigame tracker, which starts a random minigame. That way the crate code doesn't care about minigames (as it should be imo).

Reply:
Moved out the functionality out of the closet control code

However, expect this to randomly break sometime in the future when Unity decides to scramble the position of components again.